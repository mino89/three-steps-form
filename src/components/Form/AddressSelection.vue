<template>
  <form @submit.prevent="checkUser" ref="form">
    <div>
      spedizione
      <hr />
    </div>
    <fieldset>
      <label for="name">Indirizzo</label>
      <input type="text" name="name" v-model="req.address" required />
    </fieldset>
    <button v-show="$store.state.isMobile" @click="$store.commit('CHANGE_COUNT',1)" >
      Precedente
    </button>
    <button v-show="$store.state.isMobile" type="submit" ref="submit">
      Successivo
    </button>
  </form>
</template>

<script>
export default {
  name: 'AddresSelection',
  props: {
    isMobile: Boolean,
    request: Object
  },

  data () {
    return {
      req: this.request
    }
  },
  mounted () {},
  methods: {
    handleSubmit () {
      this.$refs.submit.click()
    },
    checkUser () {
      this.$store.dispatch('checkAddressData', this.req)
    }
  }
}
</script>
