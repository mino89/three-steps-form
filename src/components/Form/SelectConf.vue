<template>
  <form @submit.prevent="checkAvailability" ref="form">
    <div>
      <h2 class="title">Seleziona le opzioni</h2>
      <hr />
    </div>
    <fieldset>
      <label> Taglia </label>
      <div class="radio-group">
        <div class="radio-item" v-for="item in defaults.sizes" :key="item">
          <label :for="item">{{ item }}</label>
          <input
            type="radio"
            :id="item"
            name="size"
            :value="item"
            v-model="req.size"
            required
          />
        </div>
      </div>
    </fieldset>
    <fieldset>
      <label> Colore </label>
      <div class="radio-group">
        <div class="radio-item" v-for="item in defaults.colors" :key="item">
          <label :for="item.id">
            <div class="square" :style="{ background: item.value }"></div>
          </label>
          <input
            type="radio"
            :id="item.id"
            name="color"
            :value="item.id"
            v-model="req.color"
            required
          />
        </div>
      </div>
    </fieldset>
    <fieldset>
      <label> Emote </label>
      <div class="radio-group">
        <div class="radio-item" v-for="item in defaults.emotes" :key="item">
          <label class="emote" :for="item.id">{{ item.value }}</label>
          <input
            type="radio"
            :id="item.id"
            name="emote"
            :value="item.id"
            v-model="req.emote"
            required
          />
        </div>
      </div>
    </fieldset>
    <div class="buttons-container">
      <button v-show="isMobile" type="submit" ref="submit">
        Continua
      </button>
    </div>
  </form>
</template>

<script>
export default {
  name: 'SelectConf',
  props: {
    request: Object,
    defaults: Object
  },
  data () {
    return {
      req: this.request
    }
  },
  computed: {
    isMobile () {
      return this.$store.state.isMobile
    }
  },
  methods: {
    async handleValidation () {
      await this.$refs.submit.click()
      this.$store.commit('IS_READY_PUSH', this.$refs.form.checkValidity())
    },
    checkAvailability (e) {
      if (e.currentTarget.checkValidity() && this.isMobile) {
        this.$store.dispatch('checkConfiguration', this.req)
      }
    }
  }
}
</script>

<style lang="scss" scoped>
.radio {
  &-group {
    display: flex;
    width: 100%;
    justify-content: space-between;
  }
  &-item {
    display: flex;
    flex-direction: column;
    label {
      cursor: pointer;
      padding: 0.5rem;
    }

  }
}
.emote{
  font-size: 2.5rem;
}
.square {
  height: 24px;
  width: 24px;
  border-radius: 6px;
}
</style>
