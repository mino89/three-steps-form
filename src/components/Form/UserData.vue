<template>
  <form @submit.prevent="checkUser" ref="form">
      <div>
      <h2 class="title">Dati Personali</h2>
      <hr>
    </div>
    <fieldset>
      <label for="name">Nome</label>
      <input type="text" name="name" v-model="req.name" required />
    </fieldset>
    <fieldset>
      <label for="email">Email</label>
      <input
        type="email"
        name="email"
        v-model="req.email"
        required
      />
    </fieldset>
    <div class="buttons-container">
      <button v-show="isMobile" @click="$store.commit('CHANGE_COUNT',0)" >
        Precedente
      </button>
      <button v-show="isMobile" type="submit" ref="submit">Successivo</button>
    </div>
  </form>
</template>

<script>
export default {
  name: 'UserData',
  props: {
    request: Object
  },
  data () {
    return {
      req: this.request
    }
  },
  computed: {
    isMobile () {
      return this.$store.state.isMobile
    }
  },
  methods: {
    async handleValidation () {
      await this.$refs.submit.click()
      this.$store.commit('IS_READY_PUSH', this.$refs.form.checkValidity())
    },
    checkUser (e) {
      if (e.currentTarget.checkValidity() && this.isMobile) {
        this.$store.dispatch('checkUserData', this.req)
      }
    }
  }
}
</script>
