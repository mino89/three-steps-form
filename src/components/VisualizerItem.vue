<template>
    <span class="tshirt-background" :style="{ background: color }"></span>

  <div class="tshirt-container slot">
    <i class="tshirt-icon">{{ emote }}</i>
    <tshirt-item class="tshirt-sticky-item" :color="color" />
  </div>
</template>

<script>
import TshirtItem from './TshirtItem.vue'
export default {
  components: { TshirtItem },
  name: 'VisualizerItem',
  props: {
    config: Object,
    defaults: Object
  },
  computed: {
    dynamicBg () {
      return {
        // in the case of redComp, greenComp and blueComp are a vue prop or data
        background: 'green'
      }
    },
    size () {
      return this.config.size ? this.config.size : 'M'
    },
    color () {
      return this.config.color
        ? this.defaults.colors.find((x) => x.id === this.config.color).value
        : '#000000'
    },
    emote () {
      return this.config.emote
        ? this.defaults.emotes.find((x) => x.id === this.config.emote).value
        : 'ðŸ–Œ'
    }
  }
}
</script>

<style lang="scss" scoped>
.tshirt {
  &-container {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    width: clamp(35vw, 40vw, 35vw);
    position: relative;
  }
  &-background {
    position: absolute;
    height: 100%;
    width: 100%;
    display: flex;
    opacity: .25;
    transition: all .3s ease;
  }

  &-icon {
    position: absolute;
    font-size:  clamp(1rem, 10vw, 8rem);
    font-style: normal;
  }
}
</style>
