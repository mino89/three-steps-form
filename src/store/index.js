/* eslint-disable */
import { createStore } from 'vuex'
import axios from 'axios'

const API = "http://localhost:4000"

export default createStore({
    state: {
        count: 0,
        isMobile: false,
        defaults: {
            sizes: ["XS", "S", "M", "L", "XXL"],
            colors: [
                { id: "red", value: "#f54b42" },
                { id: "yellow", value: "#f5e642" },
                { id: "green", value: "#48f542" },
                { id: "orange", value: "#f5aa42" },
                { id: "grey", value: "#d1d1d1" },
            ],
            emotes: [
                { id: "heart", value: "❤️" },
                { id: "bolt", value: "⚡️" },
                { id: "experiment", value: "🧪" },
                { id: "sun", value: "☀️" },
                { id: "star", value: "✨" },
            ]
        },
        configuration: {
            size: null,
            color: null,
            emote: null,
        },
        userData: {
            name: null,
            email: null,
        },
        addressData: {
            address: null
        },
        message: null
    },
    mutations: {

        CHANGE_COUNT(state, count) {
            state.count = count
        },
        UPDATE_CONFIGURATION(state, configuration) {
            state.configuration = configuration
        },
        UPDATE_MESSAGE(state, message) {
            state.message = message
        },
        CLEAR_MESSAGE(state) {
            state.message = null
        },
        SWITCH_MOBILE(state, bool) {
            state.isMobile = bool
        }
    },
    actions: {
        postRequest({ commit }, { endpoint, payload, to }) {
            axios
                .post(`${API}/${endpoint}`, payload)
                .then((res) => {
                    commit("CLEAR_MESSAGE")
                    commit("CHANGE_COUNT", to);
                    console.log(res)
                })
                .catch((err) => {
                    commit('UPDATE_MESSAGE', err.response.data.message);
                })
        },
        checkConfiguration({ dispatch }, payload) {
            return dispatch('postRequest', { endpoint: 'check-availability', payload, to: 1 })
        },
        checkUserData({ dispatch }, payload) {
            return dispatch('postRequest', { endpoint: 'check-users', payload, to: 2 })
        },
        checkAddressData({ dispatch }, payload) {
            return dispatch('postRequest', { endpoint: 'check-address', payload, to: 3 })
        },
        submitData({ dispatch }, payload) {
            return new Promise((resolve, reject) => {
                axios
                    .post(`${API}/submit-data`, payload)
                    .then((response) => {
                        resolve(response)
                    })
                    .catch((error) => {
                        reject(error)
                    })
            })
        }
    }
})