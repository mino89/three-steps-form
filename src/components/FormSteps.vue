<template>
  <div class="layout">
    <div class="col layout-preview">
      <VisualizerItem :config="configuration" :defaults="defaults" />
    </div>
    <div class="col">
      <div class="slot" v-if="!success">
        <SelectConf
          :request="configuration"
          :defaults="defaults"
          ref="selectConf"
          v-show="count == 0 || !isMobile"
        />
        <UserData
          :request="userData"
          ref="userData"
          v-show="count == 1 || !isMobile"
        />
        <AddressSelection
          :request="addressData"
          ref="addressSelection"
          v-show="count == 2 || !isMobile"
        />
        <button @click="submitMultiple()" v-show="!isMobile">Continua</button>
      </div>
      <div class="slot" v-show="success">
        Ci siamo ! abbiamo inviato una mail di conferma a {{ userData.email }}
        <br />
        <button @click="reload">Continua</button>
      </div>
    </div>
  </div>
</template>
<script>
import SelectConf from './Form/SelectConf.vue'
import UserData from './Form/UserData.vue'
import AddressSelection from './Form/AddressSelection.vue'
import VisualizerItem from './VisualizerItem.vue'
import { createToaster } from '@meforma/vue-toaster'
import { mapState } from 'vuex'
import debounce from '../utils/debounce'
const toaster = createToaster({
  type: 'error'
})

export default {
  name: 'FormSteps',
  components: {
    SelectConf,
    UserData,
    AddressSelection,
    VisualizerItem
  },
  computed: {
    ...mapState([
      'count',
      'configuration',
      'userData',
      'addressData',
      'isMobile',
      'defaults'
    ])
  },
  data () {
    return {
      success: false
    }
  },
  created () {
    this.switchMobileMode()
  },
  mounted () {
    window.addEventListener(
      'resize',
      debounce(this.switchMobileMode, 500)
    )
  },
  unmounted () {
    window.removeEventListener(
      'resize',
      debounce(this.switchMobileMode, 500)
    )
  },
  methods: {
    submitMultiple () {
      this.$refs.addressSelection.handleSubmit()
      this.$refs.userData.handleSubmit()
      this.$refs.selectConf.handleSubmit()
    },

    switchMobileMode () {
      this.$store.commit(
        'SWITCH_MOBILE',
        window.innerWidth < 600
      )
      if (this.isMobile) {
        this.$store.commit('CHANGE_COUNT', 0)
      }
    },
    reload () {
      window.location.reload()
    }
  },
  watch: {
    '$store.state.message': {
      handler: function (message) {
        if (message) toaster.show(message)
      }
    },
    '$store.state.count': {
      handler: function (count) {
        if (count > 2) {
          this.$store
            .dispatch('submitData', {
              configuration: this.configuration,
              user: this.userData,
              address: this.addressData
            })
            .then(() => {
              this.success = true
            })
            .catch((error) => {
              this.$store.commit('UPDATE_MESSAGE', error.response.data.message)
            })
        }
      }
    }
  }
}
</script>

<style lang="scss" scoped>
@import "../assets/styles/variables";
.layout {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 1fr;
  &-preview {
    display: grid;
    justify-content: center;
    align-items: center;
  }
  @media all and(min-width: 600px) {
    grid-template-columns: 1fr 1fr;
    grid-template-rows: 1fr;
    min-height: 100vh;
    grid-auto-flow: column;
  }
}
.col {
  display: grid;
  align-items: center;
  position: relative;
}
</style>
