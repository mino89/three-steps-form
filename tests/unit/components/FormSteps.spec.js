
import { mount } from '@vue/test-utils'
import FormSteps from '@/components/FormSteps.vue'
import SelectConf from '@/components/Form/SelectConf.vue'
import UserData from '@/components/Form/UserData.vue'
import AddressSelection from '@/components/Form/AddressSelection.vue'
import VisualizerItem from '@/components/VisualizerItem.vue'
import store from '@/store'

describe('FormSteps.vue', () => {
  const wrapper = mount(FormSteps, {
    global: {
      plugins: [store]

    },
    components: {
      SelectConf,
      UserData,
      AddressSelection,
      VisualizerItem
    }
  })

  const SelectConfCmp = wrapper.findComponent({ name: 'SelectConf' })
  const UserDataCmp = wrapper.findComponent({ name: 'UserData' })
  const AddressSelectionCmp = wrapper.findComponent({ name: 'AddressSelection' })
  const VisualizerItemCmp = wrapper.findComponent({ name: 'VisualizerItem' })

  const FinalStep = wrapper.find('.final-step')

  const spy = jest.fn()
  wrapper.vm.submitMultiple = spy

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('must render all children components', async () => {
    expect(VisualizerItemCmp.isVisible()).toBe(true)
    expect(SelectConfCmp.isVisible()).toBe(true)
    expect(UserDataCmp.isVisible()).toBe(true)
    expect(AddressSelectionCmp.isVisible()).toBe(true)
  })

  it('must render all children components in desktop mode', async () => {
    expect(SelectConfCmp.isVisible()).toBe(true)
    expect(UserDataCmp.isVisible()).toBe(true)
    expect(AddressSelectionCmp.isVisible()).toBe(true)
  })

  it('must render component Based On steps in mobile Mode', async () => {
    await store.commit('SWITCH_MOBILE', true)
    expect(SelectConfCmp.isVisible()).toBe(true)
    expect(UserDataCmp.isVisible()).toBe(false)
    expect(AddressSelectionCmp.isVisible()).toBe(false)
  })

  it('must render second Step on Mobile', async () => {
    await store.commit('SWITCH_MOBILE', true)
    await store.commit('CHANGE_COUNT', 1)
    expect(SelectConfCmp.isVisible()).toBe(false)
    expect(UserDataCmp.isVisible()).toBe(true)
    expect(AddressSelectionCmp.isVisible()).toBe(false)
  })

  it('must render third Step on Mobile', async () => {
    await store.commit('SWITCH_MOBILE', true)
    await store.commit('CHANGE_COUNT', 2)
    expect(SelectConfCmp.isVisible()).toBe(false)
    expect(UserDataCmp.isVisible()).toBe(false)
    expect(AddressSelectionCmp.isVisible()).toBe(true)
  })

  it('must call the method to submit data', async () => {
    await store.commit('SWITCH_MOBILE', false)
    wrapper.find('.submit-form').trigger('click')
    expect(wrapper.vm.submitMultiple).toBeCalled()
  })

  it('must render final Step', async () => {
    expect(FinalStep.isVisible()).toBe(false)
    await store.commit('SWITCH_SUCCESS')
    expect(FinalStep.isVisible()).toBe(true)
  })
})
