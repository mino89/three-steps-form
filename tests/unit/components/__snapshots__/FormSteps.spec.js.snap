// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSteps.vue renders correctly 1`] = `
<div
  class="layout"
>
  <div
    class="col layout-preview"
  >
    
    <span
      class="tshirt-background"
    />
    <div
      class="tshirt-container slot"
    >
      <i
        class="tshirt-icon"
      >
        üçÉ
      </i>
      <svg
        class="tshirt-sticky-item"
        id="Capa_1"
        version="1.1"
        viewBox="0 0 295.526 295.526"
        x="0px"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
        y="0px"
      >
        <g>
          <path
            d="M147.763,44.074c12.801,0,23.858-8.162,27.83-20.169c-7.578,2.086-17.237,3.345-27.83,3.345   c-10.592,0-20.251-1.259-27.828-3.345C123.905,35.911,134.961,44.074,147.763,44.074z"
          />
          <path
            d="M295.158,58.839c-0.608-1.706-1.873-3.109-3.521-3.873l-56.343-26.01c-11.985-4.06-24.195-7.267-36.524-9.611   c-0.434-0.085-0.866-0.126-1.292-0.126c-3.052,0-5.785,2.107-6.465,5.197c-4.502,19.82-22.047,34.659-43.251,34.659   c-21.203,0-38.749-14.838-43.25-34.659c-0.688-3.09-3.416-5.197-6.466-5.197c-0.426,0-0.858,0.041-1.292,0.126   c-12.328,2.344-24.538,5.551-36.542,9.611L3.889,54.965c-1.658,0.764-2.932,2.167-3.511,3.873   c-0.599,1.726-0.491,3.589,0.353,5.217l24.46,48.272c1.145,2.291,3.474,3.666,5.938,3.666c0.636,0,1.281-0.092,1.917-0.283   l27.167-8.052v161.97c0,3.678,3.001,6.678,6.689,6.678h161.723c3.678,0,6.67-3.001,6.67-6.678V107.66l27.186,8.052   c0.636,0.191,1.28,0.283,1.915,0.283c2.459,0,4.779-1.375,5.94-3.666l24.469-48.272C295.629,62.428,295.747,60.565,295.158,58.839z   "
          />
        </g>
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
        <g />
      </svg>
    </div>
    
  </div>
  <div
    class="col"
  >
    <div
      class="slot"
    >
      <h1
        class="main-title main-title-spaces"
      >
        Configura la tua t-shirt!
      </h1>
      <div>
        <form>
          <div>
            <h2
              class="title"
            >
              Seleziona le opzioni
            </h2>
            <hr />
          </div>
          <fieldset>
            <label>
               Taglia 
            </label>
            <div
              class="radio-group"
            >
              
              <div
                class="radio-item"
              >
                <label
                  for="XS"
                >
                  XS
                </label>
                <input
                  id="XS"
                  name="size"
                  required=""
                  type="radio"
                  value="XS"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  for="S"
                >
                  S
                </label>
                <input
                  id="S"
                  name="size"
                  required=""
                  type="radio"
                  value="S"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  for="M"
                >
                  M
                </label>
                <input
                  id="M"
                  name="size"
                  required=""
                  type="radio"
                  value="M"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  for="L"
                >
                  L
                </label>
                <input
                  id="L"
                  name="size"
                  required=""
                  type="radio"
                  value="L"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  for="XXL"
                >
                  XXL
                </label>
                <input
                  id="XXL"
                  name="size"
                  required=""
                  type="radio"
                  value="XXL"
                />
              </div>
              
            </div>
          </fieldset>
          <fieldset>
            <label>
               Colore 
            </label>
            <div
              class="radio-group"
            >
              
              <div
                class="radio-item"
              >
                <label
                  for="red"
                >
                  <div
                    class="square"
                    style="background: rgb(245, 75, 66);"
                  />
                </label>
                <input
                  id="red"
                  name="color"
                  required=""
                  type="radio"
                  value="red"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  for="yellow"
                >
                  <div
                    class="square"
                    style="background: rgb(245, 230, 66);"
                  />
                </label>
                <input
                  id="yellow"
                  name="color"
                  required=""
                  type="radio"
                  value="yellow"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  for="green"
                >
                  <div
                    class="square"
                    style="background: rgb(72, 245, 66);"
                  />
                </label>
                <input
                  id="green"
                  name="color"
                  required=""
                  type="radio"
                  value="green"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  for="orange"
                >
                  <div
                    class="square"
                    style="background: rgb(245, 170, 66);"
                  />
                </label>
                <input
                  id="orange"
                  name="color"
                  required=""
                  type="radio"
                  value="orange"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  for="grey"
                >
                  <div
                    class="square"
                    style="background: rgb(209, 209, 209);"
                  />
                </label>
                <input
                  id="grey"
                  name="color"
                  required=""
                  type="radio"
                  value="grey"
                />
              </div>
              
            </div>
          </fieldset>
          <fieldset>
            <label>
               Emote 
            </label>
            <div
              class="radio-group"
            >
              
              <div
                class="radio-item"
              >
                <label
                  class="emote"
                  for="cactus"
                >
                  üåµ
                </label>
                <input
                  id="cactus"
                  name="emote"
                  required=""
                  type="radio"
                  value="cactus"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  class="emote"
                  for="pine"
                >
                  üå≤
                </label>
                <input
                  id="pine"
                  name="emote"
                  required=""
                  type="radio"
                  value="pine"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  class="emote"
                  for="palm"
                >
                  üå¥
                </label>
                <input
                  id="palm"
                  name="emote"
                  required=""
                  type="radio"
                  value="palm"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  class="emote"
                  for="sun"
                >
                  ‚òÄÔ∏è
                </label>
                <input
                  id="sun"
                  name="emote"
                  required=""
                  type="radio"
                  value="sun"
                />
              </div>
              <div
                class="radio-item"
              >
                <label
                  class="emote"
                  for="star"
                >
                  ‚ú®
                </label>
                <input
                  id="star"
                  name="emote"
                  required=""
                  type="radio"
                  value="star"
                />
              </div>
              
            </div>
          </fieldset>
          <div
            class="buttons-container"
          >
            <button
              style="display: none;"
              type="submit"
            >
               Continua 
            </button>
          </div>
        </form>
        <form>
          <div>
            <h2
              class="title"
            >
              Dati Personali
            </h2>
            <hr />
          </div>
          <fieldset>
            <label
              for="name"
            >
              Nome
            </label>
            <input
              name="name"
              required=""
              type="text"
            />
          </fieldset>
          <fieldset>
            <label
              for="email"
            >
              Email
            </label>
            <input
              name="email"
              required=""
              type="email"
            />
          </fieldset>
          <div
            class="buttons-container"
          >
            <button
              style="display: none;"
            >
               Precedente 
            </button>
            <button
              style="display: none;"
              type="submit"
            >
              Successivo
            </button>
          </div>
        </form>
        <form>
          <div>
            <h2
              class="title"
            >
              Spedizione
            </h2>
            <hr />
          </div>
          <fieldset>
            <label
              for="name"
            >
              Indirizzo
            </label>
            <input
              name="address"
              required=""
              type="text"
            />
          </fieldset>
          <div
            class="buttons-container"
          >
            <button
              style="display: none;"
            >
               Precedente 
            </button>
            <button
              style="display: none;"
              type="submit"
            >
               Successivo 
            </button>
          </div>
        </form>
        <div
          class="buttons-container"
        >
          <button
            class="submit-form"
          >
            Continua
          </button>
        </div>
      </div>
      <div
        class="final-step"
        style="display: none;"
      >
        <hr />
        <br />
         Ci siamo ! abbiamo inviato una mail di conferma a  
        <br />
        <br />
        <div
          class="buttons-container alt-class"
        >
          <button>
            Continua
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
